### ISSUE TITLE
Clarify in which Realm records, record arrays, and associated objects are created

### ISSUE THREAD

**Issue by ajklein** [arv/ecmascript-object-observe#3]
The current spec is silent on this, which will cause records to be created in the Realm of the function where object mutation occurs, record arrays to be created who-knows-where (maybe in the Realm of the first mutation per observer?).

The V8 implementation diverges from this by using the Realm V8 internally associates with the observed object for all such creations.

**Comment by arv** [comment]
@domenic suggested using the [[Realm]] of the `Object.observe` function. I'll provide a PR for that.

**Comment by ajklein** [comment]
There can be multiple `Object.observe` functions per object, just as there can be observer functions from multiple realms. I'm not clear on how @domenic's suggestion gives a clear answer about which [[Realm]] To use.

**Comment by arv** [comment]
Yeah. It is not obvious. I'm going to create a branch and see how it works out.

**Comment by domenic** [comment]
My thinking (which was pretty cursory) is that when someone does

```js
Object.observe(objFromOtherRealm, function (records) {
});
```

they expect the records to have `Object.prototype` from the current realm, i.e.

```js
assert(Object.getPrototypeOf(records[0]) === Object.prototype)
```

**Comment by arv** [comment]
Or?

```js
assert(Object.getPrototypeOf(records[0]) === callbackFunction.__proto__.__proto_)
```

the `callbackFunction` might come from a different realm too

**Comment by arv** [comment]
Another issue is.

```js
windowA.Object.getNotifier(obj) === windowB.Object.getNotifier(obj);  // ?
```

**Comment by domenic** [comment]
Between the realm of the `Object.observe` call and the realm of the callback function, I don't see a compelling preference either way. Mainly because I have a hard time imagining a reasonable scenario where they would be different. (Whereas, it's easy to imagine the scenario of observing an object from a different realm.)

So given that, I would say between those two choose whichever makes the spec and implementation easier.
